docs/docs/concepts.mdx

```diff
@@ -496,35 +501,56 @@ For specifics on how to use retrievers, see the [relevant how-to guides here](/d
 ### Tools
 <span data-heading-keywords="tool,tools"></span>
 
-Tools are interfaces that an agent, a chain, or a chat model / LLM can use to interact with the world.
+Tools are utilities designed to be called by a model: their inputs are designed to be generated by models, and their outputs are designed to be passed back to models.
+Tools are needed whenever you want a model to control parts of your code or call out to external APIs.
+
+A tool consists of:
+
+1. The name of the tool.
+2. A description of what the tool does.
+3. A JSON schema defining the inputs to the tool.
+4. A function (and, optionally, an async variant of the function).
+
+When a tool is bound to a model, the name, description and JSON schema are provided as context to the model.
+Given a list of tools and a set of instructions, a model can request to call one or more tools with specific inputs.
+Once the chosen tools are invoked, the results can be passed back to the model so that it can complete whatever task
+it's performing.
+
+#### Tool inputs
+
+A tool can take arbitrary arguments as input. At runtime, these arguments should be passed in as a dict to `tool.invoke({...})`.
+
+Alternatively, a tool can take a ToolCall as input, e.g. `tool.invoke(ToolCall(args={...}, ...))`.
+A ToolCall is what is generated by a [tool-calling model](/docs/concepts/#functiontool-calling).
+Along with the name of the tool to call and the arguments to that tool, it also contains metadata like a tool call ID, which is needed for passing the tool result back to a model.
+
+A tool also has access to the RunnableConfig that's passed into whatever chain the tool is a part of. This allows you to write tool logic that can be parameterized by the chain config. See the how-to guide for [passing in configs here](/docs/how_to/tool_configure/).
```

Comment:
nit: same as above, wrap in backticks

TODO for me is to add a concepts section on `RunnableConfig`